Modules/Sources/PointOfSale/Presentation/Item Selector/ChildItemList.swift

```
@@ -78,7 +78,9 @@ private extension ChildItemList {
             .transition(.opacity)
             .refreshable {
                 analyticsTracker.trackRefresh()
-                await itemsController.refreshItems(base: node)
+                await Task {
+                    await itemsController.refreshItems(base: node)
+                }.value
```

I thought that this is only needed because we set the state to `loading` on a refresh, which isn't really needed – the refreshing spinner that iOS puts up for us is the indicator that something's happening. However, removing the task shows that some `explicitlyCancelled` errors are coming through in that case.

I remember writing some code to swallow these – they're not real errors, it's an AlamoFire thing when a previous request gets cancelled, I think. Perhaps we need the same in the incremental sync service somewhere? With that, and removing the call to always set a loading state, I think we could get rid of the task.

---

I noticed it happening even before I had a loading state, and only had an error state. It was enough to set the error to nil before the incremental sync to cancel it.

In short, if I made any view updates while the refreshItems was still ongoing, the SwiftUI would kill the refreshable task, which caused an explicitlyCancelled error to happen. Even if we swallow the error not to be displayed, the refresh task remains canceled. 

I think the problem may be with the view structure, since it doesn't happen on the ItemListView. However, I wasn't able to find a structure that wouldn't produce this cancellation...

---

Can you reproduce it with the existing item controller? If not, perhaps that rules out the view structure?

> In short, if I made any view updates while the refreshItems was still ongoing, the SwiftUI would kill the refreshable task, which caused an explicitlyCancelled error to happen. Even if we swallow the error not to be displayed, the refresh task remains canceled.

Yes, that's unavoidable unfortunately. I'm confusing two memories, we had explicitly cancelled errors from elsewhere for something else. The key is that we can't change the view while we're refreshing.

---

> Can you reproduce it with the existing item controller? If not, perhaps that rules out the view structure?

I'll try!